config/initializers/sidekiq.rb

```diff
@@ -13,7 +10,6 @@
     config.try(:reliable!)
 
     config.redis = ThreeScale::RedisConfig.new(System::Application.config.sidekiq).config
-    config.error_handlers << System::ErrorReporting.method(:report_error)
```

Comment:
removing this, how do we ensure any errors are both - logged and reported to bugsnag?

Reply:
This was required to report to Bugsnag, but the last Bugsnag release adds support for Sidekiq 7 and the integration is transparent, so we don't need this hack anymore.

Reply:
And are we still logging the errors? `report_error` also logs them.

Reply:
I've been checking this: It logs in the console and report to Bugsnag, but the logs in the console are not different:

Before:
```
Exception -- {:exception=>{:class=>NoMethodError, :message=>"undefined method `no_method' for nil:NilClass", :backtrace=>["/home/jlledom/redhat/porta/app/workers/sphinx_account_indexation_worker.rb:12:in `perform'", "/home/jlledom/.asdf/installs/ruby/2.7.8/lib/ruby/gems/2.7.0/gems/activejob-6.0.6.1/lib/active_job/execution.rb:40:in `block in perform_now'", "/home/jlledom/.asdf/installs/ruby/2.7.8/lib/ruby/gems/2.7.0/gems/activesupport-6.0.6.1/lib/active_support/callbacks.rb:112:in `block in run_callbacks'", "/home/jlledom/.asdf/installs/ruby/2.7.8/lib/ruby/gems/2.7.0/gems/bugsnag-6.26.0/lib/bugsnag/integrations/rails/active_job.rb:38:in `block (2 levels) in included'"]}, :parameters=>{:context=>"Job raised exception", :job=>{"retry"=>true, "queue"=>"default", "class"=>"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper", "wrapped"=>"SphinxAccountIndexationWorker", "args"=>[{"job_class"=>"SphinxAccountIndexationWorker", "job_id"=>"e1e4e6fd-28c9-4b52-83b9-be32b6c89c8c", "provider_job_id"=>nil, "queue_name"=>"default", "priority"=>nil, "arguments"=>[{"_aj_serialized"=>"ActiveJob::Serializers::ModuleSerializer", "value"=>"Account"}, 3], "executions"=>0, "exception_executions"=>{}, "locale"=>"en", "timezone"=>"UTC", "enqueued_at"=>"2024-04-15T11:47:04Z"}], "jid"=>"6237b55784171ae872bad35d", "created_at"=>1713181624.713827, "enqueued_at"=>1713181927.5125213, "error_message"=>"undefined method `no_method' for nil:NilClass", "error_class"=>"NoMethodError", "failed_at"=>1713181624.7322717, "retry_count"=>3, "retried_at"=>1713181886.8906841}, :jobstr=>"{\"retry\":true,\"queue\":\"default\",\"class\":\"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper\",\"wrapped\":\"SphinxAccountIndexationWorker\",\"args\":[{\"job_class\":\"SphinxAccountIndexationWorker\",\"job_id\":\"e1e4e6fd-28c9-4b52-83b9-be32b6c89c8c\",\"provider_job_id\":null,\"queue_name\":\"default\",\"priority\":null,\"arguments\":[{\"_aj_serialized\":\"ActiveJob::Serializers::ModuleSerializer\",\"value\":\"Account\"},3],\"executions\":0,\"exception_executions\":{},\"locale\":\"en\",\"timezone\":\"UTC\",\"enqueued_at\":\"2024-04-15T11:47:04Z\"}],\"jid\":\"6237b55784171ae872bad35d\",\"created_at\":1713181624.713827,\"enqueued_at\":1713181927.5125213,\"error_message\":\"undefined method `no_method' for nil:NilClass\",\"error_class\":\"NoMethodError\",\"failed_at\":1713181624.7322717,\"retry_count\":3,\"retried_at\":1713181886.8906841}"}}
```

Now:
```
WARN: {"context":"Job raised exception","job":{"retry":true,"queue":"default","class":"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper","wrapped":"SphinxAccountIndexationWorker","args":[{"job_class":"SphinxAccountIndexationWorker","job_id":"e1e4e6fd-28c9-4b52-83b9-be32b6c89c8c","provider_job_id":null,"queue_name":"default","priority":null,"arguments":[{"_aj_serialized":"ActiveJob::Serializers::ModuleSerializer","value":"Account"},3],"executions":0,"exception_executions":{},"locale":"en","timezone":"UTC","enqueued_at":"2024-04-15T11:47:04Z"}],"jid":"6237b55784171ae872bad35d","created_at":1713181624.713827,"enqueued_at":1713183932.1412907,"error_message":"undefined method `no_method' for nil:NilClass","error_class":"NoMethodError","failed_at":1713181624.7322717,"retry_count":4,"retried_at":1713183679.3011074}}
```

Reports to Bugsnag are also different:

Before: https://app.bugsnag.com/3scale-networks-sl/system/errors/6617c2a7d6a384000858a91e?filters[error.status]=open&filters[event.since]=30d&event_id=661d14e800e6f51f0b950000

Now: https://app.bugsnag.com/3scale-networks-sl/system/errors/6617c2a7d6a384000858a91e?filters[error.status]=open&filters[event.since]=30d&event_id=661d180e00e6d9d89f420000

They are more detailed now, though they were marked as `handled` before.

I checked I could just return to the old system and change nothing. Opinions?

Reply:
Nice!

Reply:
New is fine for me.

Reply:
Yeah, that "sidekiq" tab (which I guess is added automatically) looks good.
Thanks for checking @jlledom !